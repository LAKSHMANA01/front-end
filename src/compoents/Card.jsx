import React from 'react'
import {FaBox, FaCog, FaShoppingCart, FaTachometerAlt, FaUser} from 'react-icons/fa'

const Card =({ icon, title , value})=>{
  return (
   <><div  className='bg-white  text-dark p-4 shadow-md items-center space-x-5 dark:text-red-400'>
     <div className=' text-2xl text-gray-500'>{icon}</div>
     <div className='text-lg font-semibold'><h2>{title}</h2>
     <p className=' text-xl'> {value}</p></div>
   </div>
    

   </>
  
  )
}
export default Card