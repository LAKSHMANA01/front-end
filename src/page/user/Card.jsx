import React from "react";
import {
  TrendingUp,
  TrendingDown,
  Users,
  ShoppingCart,
  DollarSign,
  Activity,
} from "lucide-react"; // Add the necessary icons

const DashboardCard = ({
  icon,
  title,
  value,
  trend = 0,
  percentage = null,
  status = "neutral",
  timeFrame = "vs last month",
  onClick,
}) => {
  // Determine the color based on the status
  const getStatusColor = () => {
    switch (status) {
      case "success":
        return "text-green-600";
      case "warning":
        return "text-amber-600";
      case "danger":
        return "text-red-600";
      default:
        return "text-blue-600";
    }
  };

  // Larger icon sizes
  const iconSize = 32; // Increase icon size for better prominence

  return (
    <div
      onClick={onClick}
      className="bg-white rounded-2xl p-6 shadow-xl hover:shadow-2xl transition-all duration-300 cursor-pointer border border-gray-100 dark:bg-gray-800 dark:border-gray-700"
    >
      {/* Icon and trend section */}
      <div className="flex items-center justify-between mb-4">
        <div className={`p-4 rounded-lg ${getStatusColor()} bg-opacity-20`}>
          {/* Display large icons */}
          <div className="flex justify-center">
            {React.cloneElement(icon, {
              size: iconSize,
              className: `${getStatusColor()}`,
            })}
          </div>
        </div>
        {/* {percentage !== null && (
          <div
            className={`flex items-center space-x-1 
            ${trend >= 0 ? "text-green-600" : "text-red-600"}`}
          >
            {trend >= 0 ? (
              <TrendingUp size={20} />
            ) : (
              <TrendingDown size={20} />
            )}
            <span className="text-sm font-semibold">{Math.abs(percentage)}%</span>
          </div>
        )} */}
      </div>

      {/* Card content */}
      <div className="space-y-2">
        <h3 className="text-lg font-semibold text-gray-700 dark:text-gray-300">
          {title}
        </h3>
        <div className="flex items-end space-x-2">
          <p className="text-3xl font-bold text-gray-800 dark:text-white">
            {value}
          </p>
          {trend !== 0 && (
            <span
              className={`text-sm pb-1 ${
                trend >= 0 ? "text-green-600" : "text-red-600"
              }`}
            >
              {trend >= 0 ? "+" : ""}
              {trend}%
            </span>
          )}
        </div>
      </div>
    </div>
  );
};

export default DashboardCard;
