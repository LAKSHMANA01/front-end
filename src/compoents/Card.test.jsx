// Card.test.jsx
import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import Card from './Card';

// Define a dummy icon component that displays its size via a data attribute.
const DummyIcon = ({ size }) => <span data-testid="icon" data-size={size}>Icon</span>;

describe('Card Component', () => {
  const title = 'Test Title';
  const value = '42';
  const onClick = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('renders title and value', () => {
    render(<Card icon={<DummyIcon />} title={title} value={value} onClick={onClick} />);
    
    // Check that the title and value appear in the document.
    expect(screen.getByText(title)).toBeInTheDocument();
    expect(screen.getByText(value)).toBeInTheDocument();
  });

  test('calls onClick when the card is clicked', () => {
    const { container } = render(
      <Card icon={<DummyIcon />} title={title} value={value} onClick={onClick} />
    );
    
    // Click the outermost div (the clickable card)
    fireEvent.click(container.firstChild);
    expect(onClick).toHaveBeenCalledTimes(1);
  });

  test('renders the icon with the correct size prop', () => {
    render(<Card icon={<DummyIcon />} title={title} value={value} onClick={onClick} />);
    
    // The DummyIcon should receive the size 32 via React.cloneElement
    const icon = screen.getByTestId('icon');
    expect(icon).toHaveAttribute('data-size', '32');
  });

  test('matches the snapshot', () => {
    const { container } = render(
      <Card icon={<DummyIcon />} title={title} value={value} onClick={onClick} />
    );
    expect(container).toMatchSnapshot();
  });
});