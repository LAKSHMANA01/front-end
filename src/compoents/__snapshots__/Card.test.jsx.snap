// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component matches the snapshot 1`] = `
<div>
  <div
    class="bg-white rounded-2xl p-6 shadow-xl hover:shadow-2xl transition-all duration-300 cursor-pointer border border-gray-100 dark:bg-gray-800 dark:border-gray-700"
  >
    <div
      class="flex items-center justify-between mb-4"
    >
      <div
        class="p-4 rounded-lg bg-opacity-20"
      >
        <div
          class="flex justify-center"
        >
          <span
            data-size="32"
            data-testid="icon"
          >
            Icon
          </span>
        </div>
      </div>
    </div>
    <div
      class="space-y-2"
    >
      <h3
        class="text-lg font-semibold text-gray-700 dark:text-gray-300"
      >
        Test Title
      </h3>
      <div
        class="flex items-end space-x-2"
      >
        <p
          class="text-3xl font-bold text-gray-800 dark:text-white"
        >
          42
        </p>
      </div>
    </div>
  </div>
</div>
`;
